src/app/app.component.ts

```diff
@@ -1419,6 +1428,57 @@ export class AppComponent implements OnInit, AfterViewInit, CanvasTableSelectLis
       this.router.navigate(['/'], { fragment });
     }
   }
-}
 
-// vim: set shiftwidth=2
+  get selectedMessageIds() {
+    return Array.from(this.selectedRows).map(x => x.id);
+  }
+
+  updateRows() {
```

Comment:
This is where I normalize the sync messages and non-sync messages. It's a mess but the simplest I could thing of